apps/omg_watcher/lib/exit_processor.ex

```diff
@@ -465,21 +467,19 @@ defmodule OMG.Watcher.ExitProcessor do
     spend_blknum
   end
 
-  defp run_block_getting(%ExitProcessor.Request{blknums_to_get: blknums} = request) do
+  defp run_block_getting(%ExitProcessor.Request{blknums_to_get: blknums} = request),
+    do: %{request | blocks_result: do_block_getting(blknums)}
+
+  defp run_ife_block_getting(%ExitProcessor.Request{piggybacked_blknums_to_get: blknums} = request),
+    do: %{request | piggybacked_blocks_result: do_block_getting(blknums)}
+
+  defp do_block_getting(blknums) do
     _ = Logger.debug("blknums_to_get: #{inspect(blknums)}")
     {:ok, hashes} = OMG.DB.block_hashes(blknums)
     _ = Logger.debug("hashes: #{inspect(hashes)}")
     {:ok, blocks} = OMG.DB.blocks(hashes)
     _ = Logger.debug("blocks_result: #{inspect(blocks)}")
-    %{request | blocks_result: blocks}
-  end
 
-  defp run_ife_block_getting(%ExitProcessor.Request{piggybacked_blknums_to_get: blknums} = request) do
-    _ = Logger.debug("piggybacked_blknums_to_get: #{inspect(blknums)}")
-    {:ok, hashes} = OMG.DB.block_hashes(blknums)
-    _ = Logger.debug("piggybacked_hashes: #{inspect(hashes)}")
-    {:ok, blocks} = OMG.DB.blocks(hashes)
-    _ = Logger.debug("piggybacked_blocks_result: #{inspect(blocks)}")
-    %{request | piggybacked_blocks_result: blocks}
+    blocks |> Enum.map(&(%Block{} = Block.from_db_value(&1)))
```

Comment:
Another not necessary type check (IMO).

Reply:
you're right, I'm not sure why I did it this way, I recall there being one spot where it made a difference, and then missed some change which made it irrelevant. Thx!